src/logic/collectibles/sources/OpenSea.ts

```diff
@@ -4,15 +4,67 @@ import { ETHEREUM_NETWORK } from 'src/logic/wallets/getWeb3'
 import NFTIcon from 'src/routes/safe/components/Balances/assets/nft_icon.png'
 import { OPENSEA_API_KEY } from 'src/utils/constants'
 
+export interface OpenSeaAssetContract {
+  address: string
+  name: string
+  image_url: string
+  symbol: string
+}
+
+export interface OpenSeaCollection {
+  name: string
+  slug: string
+}
+
+export interface OpenSeaAsset {
+  asset_contract: OpenSeaAssetContract
+  background_color: string
+  collection: OpenSeaCollection
+  description: string
+  image_thumbnail_url: string
+  name: string
+  token_id: string
+}
+
+export type OpenSeaAssets = Array<OpenSeaAsset>
+
+export interface NFTAsset {
+  address: string
+  assetContract: OpenSeaAssetContract
+  collection: OpenSeaCollection
+  description: string
+  image: string
+  name: string
+  numberOfTokens: number
+  slug: string
+  symbol: string
+}
+export type NFTAssets = Record<string, NFTAsset>
+
+export interface NFTToken {
+  assetAddress: string
+  color: string
+  description: string
+  image: string
+  name: string
+  tokenId: number | string
+}
+export type NFTTokens = Array<NFTToken>
+
+export interface Collectibles {
+  nftAssets: NFTAssets
+  nftTokens: NFTTokens
+}
+
 class OpenSea {
-  _rateLimit = async () => {}
+  _rateLimit = async (): Promise<void> => {}
 
   _endpointsUrls = {
     [ETHEREUM_NETWORK.MAINNET]: 'https://api.opensea.io/api/v1',
     [ETHEREUM_NETWORK.RINKEBY]: 'https://rinkeby-api.opensea.io/api/v1',
   }
 
-  _fetch = async (url) => {
+  _fetch = async (url: string): Promise<Response> => {
     // eslint-disable-next-line no-underscore-dangle
     return fetch(url, {
```

Comment:
why are we using fetch instead of axios library we're using everywhere else in the project?

Reply:
yes, just in OpenSea class... that's mocking the same behavior than here. I'll solve it in a separate PR, it's not critical, nor is part of this PR and just works.